include/zephyr/net/conn_mgr_connectivity.h

```diff
@@ -60,6 +65,30 @@ enum net_event_conn_cmd {
 #define NET_EVENT_CONN_IF_FATAL_ERROR					\
 	(_NET_MGMT_CONN_IF_EVENT | NET_EVENT_CONN_CMD_IF_FATAL_ERROR)
 
+/**
+ * @brief net_mgmt event raised when IPv4 connectivity becomes available
+ */
+#define NET_EVENT_CONN_IPV4_CONNECTED					\
```

Comment:
Given those events are similar to `NET_EVENT_L4_CONNECTED` etc., shouldn't they be grouped together, i. .e be specified in https://github.com/zephyrproject-rtos/zephyr/blob/0b220dd6a6d1a4320e07071d1bdc69adcbe6f65b/include/zephyr/net/net_event.h#L281
? Not only we end up with those apparently related events scattered across the codebase, but also given that they use a different event base, it won't be possible to combine those events in a single event handler.

Reply:
Yeah, it would definitely be nicer to have all of them in one event group, I agree

My reason for separating them was that, back when we were discussing how to implement `conn_mgr`, one of the issues that came up was that we felt that `NET_EVENT_L4_CONNECTED` didn't really fit well in the `NET_EVENT_L4` group. But when we tried to move it, we got pushback since it was a breaking change, and decided to just leave it there.

Personally I'm neutral to where it goes, so if you think we should just stick the new events into `NET_EVENT_L4`, I'm happy to do so. It would definitely be more convenient to have them all under one roof

Reply:
I agree with @rlubos, better to have the L4 events in one place.